metrics/telemetry/middleware.go

```diff
@@ -100,10 +100,14 @@ func NewMetricsManager(issuerURL string, databaseURL string, l logrus.FieldLogge
 		salt:         uuid.New(),
 		BuildTime:    buildTime, BuildVersion: version, BuildHash: hash,
 	}
+
+	go mm.RegisterSegment()
```

Comment:
This is a side effect of `NewMetricsHandler` which might introduce some issues during testing. Not sure if I like this here.

Reply:
I'll back it to old place as we decide to separate Prometheus and telemetry flag